aten/src/ATen/native/mkl/SparseBlasImpl.cpp

```diff
@@ -179,26 +179,53 @@ void addmm_dense_result(
   matrix_descr descrA;
   descrA.type = SPARSE_MATRIX_TYPE_GENERAL;
 
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(
-      C.scalar_type(), "addmm_out_sparse_csr_impl_mkl", [&] {
-        auto beta_ = beta.to<scalar_t>();
-        auto alpha_ = alpha.to<scalar_t>();
-
-        auto mkl_sparse_mat =
-            at::mkl::sparse::MklSparseCsrDescriptor<scalar_t>(A);
-        at::mkl::sparse::mm<scalar_t>(
-            SPARSE_OPERATION_NON_TRANSPOSE,
-            alpha_,
-            mkl_sparse_mat.descriptor(),
-            descrA,
-            order,
-            B_->data_ptr<scalar_t>(),
-            columns_C,
-            ldb,
-            beta_,
-            C_->data_ptr<scalar_t>(),
-            ldc);
-      });
+  // TODO: Presumably could use the AT_DISPATCH_SPARSE_LAYOUTS
+  // macros if MklSparseCsr/BsrDescriptor can be instantiated based using layout
```

Comment:
I find separate types more convenient and as my comment indicates there's a clear path towards cleaning up. Since you already went ahead and reimplemented it in a separate PR we'll go with your approach. I'll close this out.